node/http.go

```diff
@@ -16,94 +15,115 @@ import (
 
+func (h *httpService) Run(ctx context.Context) error {
 	errCh := make(chan error)
 	defer close(errCh)
 
 	var wg conc.WaitGroup
 	defer wg.Wait()
 	wg.Go(func() {
-		if err := httpSrv.Serve(srv.listener); !errors.Is(err, http.ErrServerClosed) {
+		if err := h.srv.Serve(h.listener); !errors.Is(err, http.ErrServerClosed) {
 			errCh <- err
 		}
 	})
 
 	select {
 	case <-ctx.Done():
-		return httpSrv.Shutdown(context.Background())
+		return h.srv.Shutdown(context.Background())
 	case err := <-errCh:
 		return err
 	}
 }
 
+func makeRPCOverHTTP(listener net.Listener, jsonrpcServer *jsonrpc.Server, log utils.SimpleLogger) *httpService {
+	httpHandler := jsonrpc.NewHTTP(jsonrpcServer, log)
+	mux := http.NewServeMux()
+	mux.Handle("/", httpHandler)
+	mux.Handle("/v0_4", httpHandler)
+	return &httpService{
+		srv: &http.Server{
+			Addr:    listener.Addr().String(),
+			Handler: mux,
+			// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
+			ReadTimeout: 30 * time.Second,
+		},
+		listener: listener,
+	}
+}
+
+func makeRPCOverWebsocket(listener net.Listener, jsonrpcServer *jsonrpc.Server, log utils.SimpleLogger) *httpService {
+	wsHandler := jsonrpc.NewWebsocket(jsonrpcServer, log)
+	mux := http.NewServeMux()
+	mux.Handle("/", wsHandler)
+	mux.Handle("/v0_4", wsHandler)
+	return &httpService{
+		srv: &http.Server{
+			Addr:    listener.Addr().String(),
+			Handler: mux,
+			// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
+			ReadTimeout: 30 * time.Second,
+		},
+		listener: listener,
+	}
+}
+
+func makeMetrics(listener net.Listener) *httpService {
+	return &httpService{
+		srv: &http.Server{
+			Addr:    listener.Addr().String(),
+			Handler: promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{Registry: prometheus.DefaultRegisterer}),
+			// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
+			ReadTimeout: 30 * time.Second,
+		},
+		listener: listener,
+	}
+}
+
+func makeGRPC(listener net.Listener, database db.DB, version string) *httpService {
+	grpcHandler := grpc.NewServer()
+	gen.RegisterKVServer(grpcHandler, junogrpc.New(database, version))
+	return &httpService{
+		srv: &http.Server{
+			Addr:    listener.Addr().String(),
+			Handler: grpcHandler,
+			// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
+			ReadTimeout: 30 * time.Second,
+		},
+		listener: listener,
+	}
+}
+
+func makePPROF(listener net.Listener) *httpService {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+	return &httpService{
+		srv: &http.Server{
+			Addr:    listener.Addr().String(),
+			Handler: mux,
+			// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
+			ReadTimeout: 30 * time.Second,
+		},
+		listener: listener,
+	}
+}
+
```

Comment:
I dont think we will ever use these in tests, right?
So can we create the listener in these functions so `node.New` has less copy-pasta?

Reply:
see https://github.com/NethermindEth/juno/pull/1087/commits/7fff1473398783135ccbe821edd6fc9a2b3c5904
Feel free to revert

Reply:
LGTM. Thanks